### DISCUSSION TITLE
Action::modal How to pass props, the pass always fails

### DISCUSSION THREAD

**Discussion by killjin** [laravel/nova-issues#5598] [Q&A] [answered]
User.php   
```php
public function actions(NovaRequest $request)
 {
      return [
          Action::modal('Preview', 'PreviewResourceModal', [
              'resourceName' => 'users',
              'resourceId' => '1',
          ])->withoutConfirmation()->showInline()
      ];
}
```
1、ajax http://127.0.0.1:8000/nova-api/users/action?action=preview&pivotAction=false&search=&filters=W3siQXBwXFxOb3ZhXFxGaWx0ZXJzXFxTZWxlY3QiOiIifV0%3D&trashed= 
return
`{"modal":"PreviewResourceModal","resourceName":"users","resourceId":"1"}`
2、show modal loading and ajax：http://127.0.0.1:8000/nova-api/undefined/undefined/preview
3、to 404

How to give PreviewResourceModal props

**Comment by killjin** [comment]
Check the code, the name of the props attribute is :data, which is incompatible with built-in components and only suitable for modal components written by yourself

**Comment by killjin** [comment]
```jsx
<component
      v-if="responseModalVisible"
      :show="responseModalVisible"
      :is="actionResponseData?.modal"
      @confirm="handleResponseModalConfirm"
      @close="handleResponseModalClose"
      :data="actionResponseData"
    />
```
`:data="actionResponseData"`
It is recommended to modify it to v-bind="actionResponseData" , which is more flexible to use

**Comment by killjin** [comment]
Personally, I suggest adding a <component is> modal component to Nova.Index. Subcomponents can share state with the Nova.Index component, which can reduce the accumulation of web pages

**Comment by crynobone** [comment, marked as answer]
This is not the correct way to use `Action::modal()`, you should instead create a custom Preview modal instead of hacking on `PreviewResourceModal` and expecting everything to just work.
